components/sharing-dialog/src/access-add/access-add.js

```diff
@@ -86,17 +158,39 @@ export const AccessAdd = ({ onAdd }) => {
                     border-radius: 5px;
                     display: flex;
                     align-items: flex-end;
-                    gap: ${spacers.dp8};
+                    justify-content: space-between;
+                }
+
+                .wrapper {
                 }
 
                 .select-wrapper {
                     flex: 1;
                 }
+                .leftWrapperData {
+                    width: 35%;
+                }
+                .leftWrapperMetadata {
+                    width: 55%;
+                }
+                .rightWrapper {
+                    margin-inline-start: 8px;
+                    display: flex;
+                    align-items: flex-end;
+                    gap: ${spacers.dp8};
+                }
+                .rightWrapperMetadata {
+                    width: 45%;
+                }
+                .rightWrapperData {
+                    width: 65%;
+                }
             `}</style>
         </>
     )
 }
 
 AccessAdd.propTypes = {
     onAdd: PropTypes.func.isRequired,
+    dataSharing: PropTypes.bool,
```

Comment:
Thank you for the reminder! I've cleaned up the type definition, which also includes some fixes:
- I've added in dataSharing on the component level
- I've updated the users/groups to be arrays of SharingObject
- I've replaced the import of `ModalOnCloseEventHandler` with the definition from ui/modal types (the type was exported from modal types and only consumed in this type file, but was imported incorrectly)

I spoke with @Birkbjo and we decided that for the access type, we would just use the new type definition `{metadata: 'ACCESS_NONE' | 'VIEW_ONLY' | 'VIEW_AND_EDIT'; data: 'ACCESS_NONE' | 'VIEW_ONLY' | 'VIEW_AND_EDIT'}` rather than also advertise the existence of the backwards compatible type. (Note: that none of our DHIS2 repos appear to currently be using the `initialSharingSettings` prop when consuming <SharingDialog> component)